pokete.py

```diff
@@ -413,287 +387,36 @@ def action(self, ob):
-{self.name + " missed!" if n_hp == 0 and attac.factor != 0 else ""}\n\
-{"That was very effective! " if effectivity == 1.3 and n_hp > 0 else ""}\
-{"That was not effective! " if effectivity == 0.5 and n_hp > 0 else ""}')
-            if enem == self:
-                time.sleep(1)
-                fightmap.outp.outp(f'{self.ext_name} hurt it self!')
-            if n_hp != 0 or attac.factor == 0:
-                attac.give_effect(enem)
-            for obj in [enem, self] if enem != self else [enem]:
-                obj.hp_bar.update(obj.oldhp)
-            logging.info("[Poke][%s] Used %s: %s", self.name, attac.name,
-                         str({"eff": effectivity, "n_hp": n_hp}))
-            self.label_rechar()
-            fightmap.show()
-
-    def evolve(self):
-        """Evolves the Pokete to its evolve_poke"""
-        if not self.player:
-            return
-        new = Poke(self.evolve_poke, self.xp, _attacks=self.attacks)
-        self.ico.remove()
-        self.ico.add(evomap, round(evomap.width / 2 - 4),
-                     round((evomap.height - 8) / 2))
-        self.moves.shine()
-        evomap.outp.outp("Look!")
-        time.sleep(0.5)
-        evomap.outp.outp(f"{evomap.outp.text}\n{self.name} is evolving!")
-        time.sleep(1)
-        for i in range(8):
-            for j, k in zip([self.ico, new.ico], [new.ico, self.ico]):
-                j.remove()
-                k.add(evomap, round(evomap.width / 2 - 4),
-                      round((evomap.height - 8) / 2))
-                time.sleep(0.7 - i * 0.09999)
-                evomap.show()
-        self.ico.remove()
-        new.ico.add(evomap, round(evomap.width / 2 - 4),
-                    round((evomap.height - 8) / 2))
-        evomap.show()
-        time.sleep(0.01)
-        new.moves.shine()
-        evomap.outp.outp(f"{self.name} evolved to {new.name}!")
-        time.sleep(5)
-        figure.pokes[figure.pokes.index(self)] = new
-        if new.identifier not in figure.caught_pokes:
-            figure.caught_pokes.append(new.identifier)
-        logging.info("[Poke] %s evolved to %s", self.name, new.name)
-        del self
-
-
 class Figure(se.Object):
-    """The figure that moves around on the map and represents the player"""
+    """The figure that moves around on the map and represents the player
+    ARGS:
+        _si: session_info dict"""
 
-    def __init__(self):
+    def __init__(self, _si):
         super().__init__("a", state="solid")
-        self.__money = 10
-        self.inv = {"poketeballs": 10}
-        self.name = ""
-        self.pokes = []
-        self.caught_pokes = []
-        self.visited_maps = []
-        self.last_center_map = ob_maps["playmap_1"]
-        self.oldmap = ob_maps["playmap_1"]
+        self.__money = _si.get("money", 10)
+        self.inv = _si.get("inv", {"poketeballs": 10})
+        self.name = _si.get("user", "DEFAULT")
+        self.pokes = [Poke((_p:=_si["pokes"][poke])["name"], _p["xp"],
```

Comment:
PEP8: E225: Missing whitespace around operators
```suggestion
        self.pokes = [Poke((_p := _si["pokes"][poke])["name"], _p["xp"],
```